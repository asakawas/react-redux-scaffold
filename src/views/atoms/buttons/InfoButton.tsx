import React, { FunctionComponent } from 'react';
import IconButton, { IconButtonProps } from '@material-ui/core/IconButton';
import InfoIcon from '@material-ui/icons/Info';

/**
 * The type of props of InfoButton.
 */
export type InfoButtonProps = Readonly<{
  /** The event handler called when the info button is clicked */
  onInfoButtonClick: IconButtonProps['onClick'];
}>;

const InfoButton: FunctionComponent<InfoButtonProps> = ({ onInfoButtonClick }) => (
  <IconButton onClick={onInfoButtonClick} size="small" aria-label="info button">
    <InfoIcon color="primary" />
  </IconButton>
);

export default InfoButton;
