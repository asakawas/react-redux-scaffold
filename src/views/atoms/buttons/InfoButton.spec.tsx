import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import InfoButton from './InfoButton';

describe('InfoButton', () => {
  test('is rendered as expected', () => {
    const { asFragment } = render(<InfoButton onInfoButtonClick={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("calls the passed handler when it's clicked", () => {
    const handler = jest.fn();
    const renderResult = render(<InfoButton onInfoButtonClick={handler} />);

    const tree = renderResult.getByLabelText('info button');
    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(1);

    fireEvent.click(tree);
    expect(handler).toHaveBeenCalledTimes(2);
  });
});
