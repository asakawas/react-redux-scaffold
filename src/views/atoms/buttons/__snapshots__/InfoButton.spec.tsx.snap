// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoButton is rendered as expected 1`] = `
<DocumentFragment>
  <button
    aria-label="info button"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
